core_lib/src/structure/filemanager.cpp

```diff
@@ -547,10 +569,19 @@ QString FileManager::backupPreviousFile(const QString& fileName)
         return "";
 
     QFileInfo info(fileName);
-    QString sBackupFile = info.completeBaseName() + ".backup." + info.suffix();
+    int backupCount = 1;
+    QDir directory(info.absoluteDir());
+    QString backupString = "backup";
+    for (QFileInfo info : directory.entryInfoList(QDir::Filter::Files)) {
+        if (info.completeBaseName().contains(backupString)) {
+            backupCount++;
+        }
+    }
+
+    QString sBackupFile = info.baseName() + "." + backupString + QString::number(backupCount) + "." + info.suffix();
```

Comment:
`completeBaseName()` instead of `baseName()`? Just in case there are dots in filenames.

Reply:
Fixed